packages/functional-tests/tests/settings/changeEmail.spec.ts

```diff
@@ -3,105 +3,118 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import { test, expect } from '../../lib/fixtures/standard';
+import { SettingsPage } from '../../pages/settings';
+import { ChangePasswordPage } from '../../pages/settings/changePassword';
+import { SecondaryEmailPage } from '../../pages/settings/secondaryEmail';
 
 const AGE_21 = '21';
-let newEmail;
 
 test.describe('severity-1 #smoke', () => {
   test.describe('change primary email tests', () => {
-    test.beforeEach(
-      async ({ credentials, pages: { settings, secondaryEmail } }) => {
-        test.slow();
-        await settings.goto();
-        await settings.secondaryEmail.clickAdd();
-        newEmail = credentials.email.replace(/(\w+)/, '$1_alt');
-        await secondaryEmail.addAndVerify(newEmail);
-        await settings.waitForAlertBar();
-        await settings.secondaryEmail.clickMakePrimary();
-      }
-    );
+    test.beforeEach(async () => {
+      test.slow();
+    });
 
     test('change primary email and login', async ({
       credentials,
       page,
-      pages: { login, settings },
+      pages: { login, settings, secondaryEmail },
     }) => {
+      await settings.goto();
+
+      const newEmail = credentials.email.replace(/(\w+)/, '$1_alt');
+      await addSecondaryEmail(settings, secondaryEmail, newEmail);
+
       await settings.signOut();
 
       // Sign in with old primary email fails
       await login.setEmail(credentials.email);
       await page.locator('button[type=submit]').click();
       await login.setPassword(credentials.password);
       await page.locator('button[type=submit]').click();
```

Comment:
We can't reuse this method because of the waitForNavigation logic on the final submit button. I'm intentionally avoiding the LoginPage btw, I think that's better left to an independent refactor.